include/revng/PipeboxCommon/LLVMContainer.h

```diff
@@ -22,12 +22,14 @@ class LLVMRootContainer {
   static constexpr llvm::StringRef MimeType = "application/x.llvm.bc";
 
 private:
-  llvm::LLVMContext Context;
+  bool Expiring = false;
+  std::unique_ptr<llvm::LLVMContext> Context;
   std::unique_ptr<llvm::Module> Module;
```

Comment:
```c++
struct ModuleAndContext {
  llvm::LLVMContext Context;
  llvm::Module Module;
};
std::optional<ModuleAndContext> TheModuleAndContext;

Reply:
"naked" modules (e.g. without `std::unique_ptr`) are really hard to handle because they are not copyable or movable, and all the LLVM infrastructure (e.g. `parseIR`, `cloneModule`) always returns a `std::unique_ptr<llvm::Module>`

Reply:
Mh, OK, maybe leave a comment.